core/vm/memory.go

```diff
@@ -66,28 +66,26 @@ func (m *Memory) Resize(size uint64) {
 }
 
 // GetCopy returns offset + size as a new slice
-func (m *Memory) GetCopy(offset, size int64) (cpy []byte) {
+func (m *Memory) GetCopy(offset, size uint64) (cpy []byte) {
 	if size == 0 {
 		return nil
 	}
 
-	if len(m.store) > int(offset) {
+	if uint64(len(m.store)) > offset {
```

Comment:
This check seems to not be needed. We always expand the memory before trying to access it, so this check is useless (plus it was always wrong because it should have been offset + size). Might as well delete it now.